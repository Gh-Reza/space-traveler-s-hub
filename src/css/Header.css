.logoimage img {
  height: 4rem;
}

.navbar {
  padding: 1.5rem;
  margin-bottom: 1rem;
  border-bottom: 1px solid gray;
}

.navbar,
.menuitems {
  display: flex;
  flex-direction: row;
  align-items: baseline;
  gap: 3rem;
}

.title-logo {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  align-items: center;
  margin-left: 2rem;
}

.menuitems {
  margin-left: auto;
  margin-right: 2rem;
}

.active-rout {
  color: blue;
}

.unactive {
  color: blue;
  text-decoration: none;
}

.myprofile::before {
  content: "|";
  padding-right: 1.5rem;
  color: rgb(70, 69, 69);
}
