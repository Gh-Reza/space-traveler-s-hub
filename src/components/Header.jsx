import { NavLink } from 'react-router-dom';
import logo from '../planet.png';
import '../css/Header.css';

const Header = () => (
  <nav className="navbar">
    <div className="title-logo">
      <span className="logoimage">
        <img src={logo} alt="planet-icon" />
      </span>
      <h1>Space Travelers&apos; Hub</h1>
    </div>
    <div className="menuitems">
      <span className="menu">
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active-rout' : 'unactive')}
        >
          Rockets
        </NavLink>
      </span>
      <span className="menu">
        <NavLink
          to="/missions"
          className={({ isActive }) => (isActive ? 'active-rout' : 'unactive')}
        >
          Missions
        </NavLink>
      </span>
      <span className="menu">
        <NavLink
          to="/profile"
          className={({ isActive }) => (isActive ? 'myprofile active-rout' : 'myprofile unactive')}
        >
          My profile
        </NavLink>
      </span>
    </div>
  </nav>
);

export default Header;
