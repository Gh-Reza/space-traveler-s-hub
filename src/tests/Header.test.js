import React from 'react';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

describe('Header Component Unit Test', () => {
  test('Header Component renders correctly', () => {
    const tree = renderer.create(<Router><Header /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Header Component Contains Title and Menu', () => {
    render(
      <Router>
        <Header />
      </Router>
    );

    expect(screen.getByText("Space Travelers' Hub")).toBeInTheDocument();
    expect(screen.getByText('Rockets')).toBeInTheDocument();
    expect(screen.getByText('Missions')).toBeInTheDocument();
    expect(screen.getByText('My profile')).toBeInTheDocument();
  });
});