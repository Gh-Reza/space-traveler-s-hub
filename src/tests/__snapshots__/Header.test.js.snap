// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Unit Test Header Component renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="title-logo"
  >
    <span
      className="logoimage"
    >
      <img
        alt="planet-icon"
        src="planet.png"
      />
    </span>
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <div
    className="menuitems"
  >
    <span
      className="menu"
    >
      <a
        aria-current="page"
        className="active-rout"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </span>
    <span
      className="menu"
    >
      <a
        className="unactive"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </span>
    <span
      className="menu"
    >
      <a
        className="myprofile unactive"
        href="/profile"
        onClick={[Function]}
      >
        My profile
      </a>
    </span>
  </div>
</nav>
`;
